features/lwipstack/lwip/src/apps/altcp_tls/altcp_tls_mbedtls.c

```diff
@@ -714,12 +728,25 @@ altcp_tls_create_config(int is_server, int have_cert, int have_pkey, int have_ca
 
   mbedtls_ssl_config_init(&conf->conf);
   mbedtls_entropy_init(&conf->entropy);
-  mbedtls_ctr_drbg_init(&conf->ctr_drbg);
+#if defined(MBEDTLS_CTR_DRBG_C)
+  mbedtls_ctr_drbg_init(&conf->_drbg);
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+  mbedtls_hmac_drbg_init(&conf->_drbg);
```

Comment:
No, we don't want to pollute code with dead code

Reply:
You don't need an `#error`. Without a DRBG the build will die on its own.

Reply:
@sbutcher-arm ah ok so that was what I was getting at, what you are confirming then is that when compilation hits this point one of those flags is guaranteed otherwise the build would already have failed? In which case I would swap the #elif for #else and comment what that default signifies

Reply:
Ok - I've been thinking about this over the weekend, and I don't think there is a valid default at the moment. Historically, we've made the default the HMAC DRBG, but with Client Lite (which is the purpose of this PR), we're moving to CTR DRBG as it has a smaller footprint.

So depending on use-case, the default is different.

Honestly, I don't think there is a valid value for a default `#else` clause, and I think the best option is what Antti has here.

*If* neither are selected, the build will not get to this point and will break in the `check_config.h` header where the DRBG configuration is checked.

Hope this satisfies you!

Reply:
We could be sure by adding an `#else` case in defensive coding style, without adding dead code or impacting binary size. This would be purely defensive coding, as the build is likely to fail anyway.

```C
#if defined(MBEDTLS_CTR_DRBG_C)
  mbedtls_ctr_drbg_init(&conf->_drbg);
#elif defined(MBEDTLS_HMAC_DRBG_C)
  mbedtls_hmac_drbg_init(&conf->_drbg);
#else
#  error Unexpected Mbed TLS configuration
#endif

Reply:
That stops people providing their own DRBG with the same function signatures.

I don't know if anybody *does* but the modular design of Mbed TLS is such that people could if they wanted to do that.

I also have no idea if it would work (as we don't test for it), but implementing the design poorly is a bit different from contradicting it.

Reply:
Fixed